### PR TITLE
opt: add branch builder push_chunk

### PR DESCRIPTION
**Pull request by gabriele-0201**

### CODE REVIEW

**Comment by gabriele-0201** [comment]
* **#569** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/569?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#585** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/585?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/585?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `master`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: nomt/src/beatree/branch/node.rs [hunk 1]
```diff
@@ -294,11 +388,281 @@ impl BranchNodeBuilder {
...
+        for (i, key) in keys.iter_mut().enumerate() {
+            key[0] = 0x11;
+            if i != 0 {
+                key[1] = 128;
+                key[2] = i as u8;
+            }
+        }
+
+        // prefix: 00010001_
+        // key 0:           _000...            // cell_poiter: 0
```
Comment (rphmeier) [inline nomt/src/beatree/branch/node.rs:607]: typo here in tests, should say `cell_pointer` here and everywhere.

**Comment by rphmeier** [comment]
I feel like I reviewed this before and the comments got lost. Is that possible?

**Comment by rphmeier** [comment]
### Merge activity

* **Dec 5, 3:42 PM EST**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/thrumdev/nomt/585).

File: nomt/src/beatree/branch/node.rs [hunk 2]
```diff
@@ -107,10 +114,31 @@ impl BranchNode {
...
     }
 
+    fn cells_mut(&mut self) -> &mut [[u8; 2]] {
+        unsafe {
+            std::slice::from_raw_parts_mut(
+                self.page[BRANCH_NODE_HEADER_SIZE..BRANCH_NODE_HEADER_SIZE + 2].as_ptr()
+                    as *mut [u8; 2],
+                self.n().into(),
+            )
+        }
```
Comment (pepyakin) [inline nomt/src/beatree/branch/node.rs:124]: What is the safety justification here?

**Pull request merged**